src/python/ASO/Rucio/Actions/MonitorLockStatus.py

```diff
@@ -110,6 +112,49 @@ def registerToPublishContainer(self, fileDocs):
             f['dataset'] = tmpLFN2DatasetMap[f['name']]
         return tmpFileDocs
 
+    def registerToMutiPubContainers(self, fileDocs):
+        """
+        Register replicas to it own publish container.
+
+        In example, we have
+        - output dataset name `/GenericTTbar/cmsbot-integration-1/USER`
+        - 2 output files:
+          - `output.root` (EDM)
+          - `myfile.txt` (Misc)
+        All `output_{job_id}.root` will registering to `/GenericTTbar/cmsbot-integration-1__output.root/USER`.
+        All `myfile_{job_id}.txt` will registering to `/GenericTTbar/cmsbot-integration-1__myfile.txt/USER`.
+
+        :param fileDocs: replicas info return from `checkLockStatus` method.
+        :type fileDocs: list of dict (fileDoc)
+
+        :return: replicas info with updated dataset name.
+        :rtype: list of dict
+        """
+        r = RegisterReplicas(self.transfer, self.rucioClient, None)
+        publishContainerFileDocs = []
+        groupFileDocs = {}
+        # Group by filename
+        for fileDoc in fileDocs:
+            filename = parseFileNameFromLFN(fileDoc['name'])
+            if filename in groupFileDocs:
+                groupFileDocs[filename].append(fileDoc)
+            else:
+                groupFileDocs[filename] = [fileDoc]
+        # Register to its own publish container
+        for filename, fileDocsInGroup in groupFileDocs.items():
+            container = ''
+            for c in self.transfer.multiPubContainers:
```

Comment:
well..wel... is it not very likely that someone puts `__` as a separator in their crab.Data.outputDatasetTag, but this still looks fragile. If this Continers name are only used for Rucio, why not adding the file name at the end ? Does Rucio enforces a list of dataTiers like DBS does ?
Or am I worried w/o reason and since we are sure that we added `__name` at end right end of ProcessedDatasetName, we can rely on the `rsplit` to do the right thing ?

Reply:
> Does Rucio enforces a list of dataTiers like DBS does ?

Yes (not 100% sure, I have tried it when I develop the Transfer Container).

> since we are sure that we added __name at end right end of ProcessedDatasetName, we can rely on the rsplit to do the right thing ?

I agree. Will change.
I came up with this approach before I learned how we resolve the original filename in PostJob/cmscp.